src/vaults/VaultManager.ts

```diff
@@ -985,7 +978,7 @@ class VaultManager {
     tran: DBTransaction,
   ): Promise<Buffer> {
     let key: Buffer | undefined;
-    key = await tran.get([...this.vaultsDbPath, 'key'], true);
+    key = await tran.getForUpdate([...this.vaultsDbPath, 'key'], true);
```

Comment:
Same with this. Doesn't seem necessary. What is the race condition that we are guarding against here? Remember it's only necessary during "write skews".

Reply:
I was only checking for if a key was updated following a read. I this case we write a key if it didn't already exist. If you feel it's unnecessary here then I can remove it.

Reply:
Yep remove, it's only for write skews. https://github.com/MatrixAI/Polykey/pull/419#discussion_r951086979 (unless this itself is a write skew!?).

Reply:
This has been removed.